org.alloytools.alloy.classic.provider/src/main/java/org/alloytools/alloy/classic/provider/AlloyClassicFacade.java

```diff
@@ -0,0 +1,365 @@
+			for (AlloySolverFactory factory : Services.getServices(AlloySolverFactory.class)) {
+				solvers.addAll(factory.getAvailableSolvers(this));
+			}
+		}
+		return solvers;
+	}
+
+	@Override
+	public Optional<Solver> getSolver(String id) {
+		return getSolvers().stream()
+			.filter(s -> s.getId()
+				.equals(id))
+			.findAny();
+	}
+
+	@Override
+	public Compiler compiler(SourceResolver resolver) {
+		return new Compiler() {
+
+			@Override
+			public Module compile(File file) {
+				try {
+					return compileSource(new String(Files.readAllBytes(file.toPath()), "UTF-8"));
+				} catch (IOException e) {
+					e.printStackTrace();
+					throw new RuntimeException(e);
+				}
+			}
+
+			@Override
+			public Module compileSource(String source) {
+				return compileSource(null, source);
+			}
+
+			Module compileSource(String path, String source) {
+				List<Option> options = getOptions(source);
+
+				A4Reporter reporter = new A4Reporter();
+
+				try {
+					CompModule module = CompUtil.parseEverything_fromString(reporter, source);
+					return new AlloyModuleClassic() {
+
+						@Override
+						public Set<TSig> getSigs() {
+							ConstList<Sig> sigs = module.getAllReachableSigs();
+							return new HashSet<>(sigs);
+						}
+
+						@Override
+						public Optional<TSig> getSig(String name) {
+							return getSigs().stream()
+								.filter(s -> s.getName()
+									.equals(name))
+								.findAny();
+						}
+
+						@Override
+						public List<TRun> getRuns() {
+							Module THIS = this;
+							return module.getAllCommands()
+								.stream()
+								.filter(c -> !c.isCheck())
+								.map(r -> (TRun) new AbstractCommand(this, r))
+								.collect(Collectors.toList());
+						}
+
+						@Override
+						public List<TCheck> getChecks() {
+							Module THIS = this;
+							return module.getAllCommands()
+								.stream()
+								.filter(c -> c.isCheck())
+								.map(r -> (TCheck) new AbstractCommand(this, r))
+								.collect(Collectors.toList());
+						}
+
+						@Override
+						public CompModule getOriginalModule() {
+							return module;
+						}
+
+						@Override
+						public Optional<String> getPath() {
+							return Optional.ofNullable(path);
+						}
+
+						@Override
+						public List<CompilerMessage> getWarnings() {
+							return Collections.emptyList();
+						}
+
+						@Override
+						public List<CompilerMessage> getErrors() {
+							return Collections.emptyList();
+						}
+
+						@Override
+						public boolean isValid() {
+							return getErrors().isEmpty();
+						}
+
+						@Override
+						public String toString() {
+							return module.toString();
+						}
+
+						@Override
+						public Map<String, String> getSourceOptions(TCommand command) {
+							return extractOptions(options, command);
+						}
+
+						@Override
+						public Compiler getCompiler() {
+							return compiler();
+						}
+					};
+				} catch (Exception e) {
+					return new AlloyModuleClassic() {
```

Comment:
This should be extracted into a class, and then the same class can be used here as a result that denotes an error an up above where it actually contains a module